packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -775,53 +736,64 @@ export class TokenRatesController extends StaticIntervalPollingController<TokenR
     tokenAddresses: Hex[];
     nativeCurrency: string;
   }): Promise<ContractMarketData> {
-    const [
-      contractExchangeInformations,
-      fallbackCurrencyToNativeCurrencyConversionRate,
-    ] = await Promise.all([
-      this.#fetchAndMapExchangeRatesForSupportedNativeCurrency({
+    const nativeTokenAddress = getNativeTokenAddress(chainId);
+
+    // Step -1: First fetch native token priced in USD
+    const nativeTokenPriceMap =
+      await this.#fetchAndMapExchangeRatesForSupportedNativeCurrency({
+        tokenAddresses: [] as Hex[], // special-case: returns only native token
+        chainId,
+        nativeCurrency: 'usd',
+      });
+
+    // Step -2: Then fetch all tracked tokens priced in USD
+    const tokenPricesInUSD =
+      await this.#fetchAndMapExchangeRatesForSupportedNativeCurrency({
         tokenAddresses,
         chainId,
-        nativeCurrency: FALL_BACK_VS_CURRENCY,
-      }),
-      getCurrencyConversionRate({
-        from: FALL_BACK_VS_CURRENCY,
-        to: nativeCurrency,
-      }),
-    ]);
-
-    if (fallbackCurrencyToNativeCurrencyConversionRate === null) {
+        nativeCurrency: 'usd',
+      });
```

Comment:
I don't think you need to make 2 calls.

The second one will automatically include the native currency in the result.

Reply:
Which by the way is something I will be removing soon, as that means fetching the native currency rate on every batch if there are too many tokens.